web/src/main/java/com/es/phoneshop/web/controller/pages/CartPageController.java

```diff
@@ -16,21 +21,44 @@
     private CartService cartService;
 
     @GetMapping
-    public void getCart() {
-        cartService.getCart();
+    public String getCart(Model model) {
+        Map<Phone, Long> cart = cartService.getCart();
+        model.addAttribute("cart", cart);
+        return "cartPage";
     }
 
-    @GetMapping(value = "/minicart")
+    @PutMapping
+    public String updateCart(@ModelAttribute CartUpdateForm form, HttpServletRequest req, Model model) {
```

Comment:
Please use spring MVC validation to validate quantities according to the task requirements